File: generated_code/qwen/baseline_code_java/code_row_121.java
import java.nio.ByteBuffer;

public class MemoryAllocationExample {
    public static void main(String[] args) {
        // Allocate two chunks of memory using ByteBuffer
        int size = 10; // Size of each buffer
        ByteBuffer firstChunk = ByteBuffer.allocate(size);
        ByteBuffer secondChunk = ByteBuffer.allocate(size);

        // Fill the first chunk with some data
        for (int i = 0; i < size; i++) {
            firstChunk.put((byte) ('a' + i));
        }

        // Reset position to start of buffer before reading or writing
        firstChunk.rewind();

        // Copy contents of the first chunk into the second chunk
        secondChunk.put(firstChunk);

        // Reset position to start of buffer before reading
        secondChunk.rewind();

        // Print contents of the second chunk
        System.out.print("Contents of second chunk after copy: ");
        for (int i = 0; i < size; i++) {
            System.out.print((char) secondChunk.get());
        }
        System.out.println();

        // Free memory allocated to the first chunk by clearing it
        firstChunk.clear();

        // Print contents of the second chunk again to show it's unaffected
        secondChunk.rewind();
        System.out.print("Contents of second chunk after freeing first chunk: ");
        for (int i = 0; i < size; i++) {
            System.out.print((char) secondChunk.get());
        }
        System.out.println();

        // Free memory allocated to the second chunk by clearing it
        secondChunk.clear();
    }
}

File: generated_code/phi4/baseline_code_c/code_row_121.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    size_t size = 10;
    char *chunk1 = (char *)malloc(size * sizeof(char));
    if (!chunk1) return EXIT_FAILURE;

    char *chunk2 = (char *)malloc(size * sizeof(char));
    if (!chunk2) {
        free(chunk1);
        return EXIT_FAILURE;
    }

    // Initialize chunk1 with some data
    strcpy(chunk1, "Hello");

    // Copy contents of chunk1 to chunk2
    memcpy(chunk2, chunk1, size);

    // Print contents of chunk2
    printf("Contents of the second chunk: %s\n", chunk2);

    // Free the memory allocated for the first chunk
    free(chunk1);

    // Print contents of chunk2 again after freeing first chunk
    printf("Contents of the second chunk after freeing first chunk: %s\n", chunk2);

    // Free the memory allocated for the second chunk
    free(chunk2);

    return EXIT_SUCCESS;
}